src/components/support-search.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react'
+import { iconUrl } from '../util/icon-url'
+import useDebounce from './use-debounce'
+
+const searchWidgets = (input) => fetch(`/api/admin/widget_search/${input}`)
+
+const SupportSearch = ({onClick = () => {}}) => {
+	const [searchText, setSearchText] = useState('')
+	const [searchResults, setSearchResults] = useState([])
+	const [isSearching, setIsSearching] = useState(false)
+	const [showDeleted, setShowDeleted] = useState(false)
+	const debouncedSearchTerm = useDebounce(searchText, 500)
+
+	useEffect(() => {
+		// Make sure we have a value (user has entered something in input)
+		if (debouncedSearchTerm) {
+
+			if (debouncedSearchTerm !== "")
+				setIsSearching(true)
+			
+			searchWidgets(debouncedSearchTerm)
+				.then(resp => {
+					// no content
+					if(resp.status === 502 || resp.status === 204) {
+						setIsSearching(false)
+						return []
+					}
```

Comment:
duplicate comment, but we should consolidate a universal rest call interface & error handling.